### PR TITLE
Sc 11779/members db

### PR DESCRIPTION
**Pull request by daniellemaxwell**
### Scope of changes

This adds the Members database that the TenantAPI will connect to for server-side requests. It mimics the [tenants db](https://github.com/rotationalio/ensign/blob/main/pkg/tenant/db/tenants.go). Tenants db tests may be viewed [here.](https://github.com/rotationalio/ensign/blob/main/pkg/tenant/db/tenants_test.go)

Approval of this PR will allow server-side handler implementation to begin for the members resource.  

Fixes SC-11779

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [x] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

Describe how reviewers can test this change to be sure that it works correctly. Add a checklist if possible.

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [x]  I have added new test fixtures as needed to support added tests
- [ ]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.

### CODE REVIEW

File: pkg/tenant/db/members.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	member.Created = time.Now()
+	member.Modified = member.Created
+
+	if err = Put(ctx, member); err != nil {
+		return err
+	}
+	return nil
+}
+
+// RetrieveMember gets a member from the data with a given id.
```
Comment (daniellemaxwell) [inline pkg/tenant/db/members.go:60]: A function will be added to list all members assigned to a tenant.

File: pkg/tenant/db/members.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+	// Check if memberID exists and return a missing
+	// id error response if it does not.
+	if member.ID.Compare(ulid.ULID{}) == 0 {
+		return ErrMissingID
+	}
+
+	if err = Put(ctx, member); err != nil {
+		return err
+	}
```
Comment (DanielSollis) [inline pkg/tenant/db/members.go:84]: Do you need to update member.Modified here or will Put() do that for you?

File: pkg/tenant/db/members_test.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+		return &pb.PutReply{
+			Success: true,
+		}, nil
+	}
+
+	err := db.UpdateMember(ctx, member)
+	require.NoError(err, "could not update member")
+
+	require.Equal(ulid.MustParse("01GKKYAWC4PA72YC53RVXAEC67"), member.ID, "member ID should not have changed")
```
Comment (DanielSollis) [inline pkg/tenant/db/members_test.go:144]: you might want to test that member.Modified has been updated.

**Review by DanielSollis — APPROVED** [review]
Looks good! I just had a quick question and found a few small typos.

File: pkg/tenant/db/members.go (hunk 2 above)
Comment (daniellemaxwell) [reply pkg/tenant/db/members.go:84]: Good catch!

**Pull request merged**